tools/lint.sh

```diff
@@ -35,14 +35,18 @@ cargo install --version 1.0.9 --locked cargo-sort
 
 # We want to check with --all-targets since it checks test code, but that flag
 # leads to build errors in enclave workspaces, so check it here.
+cargo sort --workspace --grouped $CHECK
 cargo clippy --all --all-features --all-targets -- -D warnings
+cargo fmt -- --unstable-features $CHECK
 
-for toml in $(grep --exclude-dir cargo --exclude-dir rust-mbedtls --include=Cargo.toml -r . -e '\[workspace\]' | cut -d: -f1); do
+# Run on all the nested workspaces.
+# We exclude `cargo` paths as these are crates from `[patch.crates-io]`
+for toml in $(find . -mindepth 2 -type f -not -path '*/cargo/*' -name Cargo.toml -exec grep -l -e '\[workspace\]' {} +); do
   pushd $(dirname $toml) >/dev/null
   echo "Linting in $PWD"
   cargo sort --workspace --grouped $CHECK
-  cargo fmt -- --unstable-features $CHECK
   cargo clippy --all --all-features -- -D warnings
+  cargo fmt -- --unstable-features $CHECK
```

Comment:
I wonder why you moved the fmt after clippy? Its usually faster so to me it would make sense to have it run before the slower test

Reply:
I'll move back. I think this was left over from me trying to get the `--fix` flag of clippy working. When fixing with clippy it will leave files in a state that still need formatting.

Reply:
fmt is now prior to clippy